### PR TITLE
Refactor the modals to render based on context

### PR DESCRIPTION
**Pull request by tiagoalvesdulce**
This PR aims to improve the way we manage modals right now by:

1. Render based on context.
2. Modals will no longer appear in the DOM if they are not being used (they will be completely unmounted instead of just hidden).
3. Improve Modal management and reduce boilerplate.

Closes https://github.com/decred/politeiagui/issues/1835

### CODE REVIEW

**Comment by victorgcramos** [comment]
Very nice! 🐐 👏

File: src/containers/User/Detail/Account/AdminAccount.jsx [hunk 1]
```diff
@@ -225,71 +271,12 @@ const AdminAccount = ({
...
-            {isdeactivated ? "deactivated" : "activated"}.
-          </Text>
-        }
-      />
-      <ModalChangePassword
-        onChangePassword={onChangePassword}
-        validationSchema={changePasswordValidationSchema}
-        show={showPasswordModal}
-        onClose={closePasswordModal}
-      />
```
Comment (victorgcramos) [inline src/containers/User/Detail/Account/AdminAccount.jsx:292]: very nice!!

File: src/containers/Comments/helpers.js [hunk 2]
```diff
@@ -52,3 +52,15 @@ export const sortComments = (sortOption, comments) => {
...
   return sorter(comments);
 };
+
+/**
+ * This function use currying to efficiently handle comment censoring
+ * @param {int} id
+ * @param {string} reason
+ */
+export function handleCommentCensoringInfo(cb, ...args) {
+  console.log(args.id);
```
Comment (amassarwi) [inline src/containers/Comments/helpers.js:62]: console

File: src/containers/User/Detail/Account/AdminAccount.jsx (hunk 1 above)
Comment (amassarwi) [reply src/containers/User/Detail/Account/AdminAccount.jsx:292]: neat 👏

File: src/containers/User/Detail/Account/hooks.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export function useChangePasswordModal() {
```
Comment (amassarwi) [inline src/containers/User/Detail/Account/hooks.js:1]: Is this used anywhere ? or is it part of WIP ?

Comment (tiagoalvesdulce) [reply src/containers/User/Detail/Account/hooks.js:1]: Ooops, forgot this one

**Pull request merged**